packages/core/src/external/carequality/ihe-gateway-v2/__tests__/process-xcpd.test.ts

```diff
@@ -0,0 +1,88 @@
+import fs from "fs";
+import path from "path";
+import { processXCPDResponse } from "../process-xcpd-response";
+import {
+  outboundPatientDiscoveryRespSuccessfulSchema,
+  outboundPatientDiscoveryRespFaultSchema,
+} from "@metriport/ihe-gateway-sdk";
+
+import { outboundRequest, expectedResponse } from "./constants";
+
+if (!outboundRequest.gateways[0]) {
+  throw new Error("Gateway must be provided");
+}
+const gateway = {
+  url: outboundRequest.gateways[0].url,
+  oid: outboundRequest.gateways[0].oid,
+  id: outboundRequest.gateways[0].id,
+};
+
+describe("processXCPDResponse", () => {
+  it("should process the match XCPD response correctly", async () => {
+    const xmlString = fs.readFileSync(path.join(__dirname, "xcpd_match.xml"), "utf8");
+
+    const response = processXCPDResponse({
+      xmlStringOrError: xmlString,
+      outboundRequest,
+      gateway,
+    });
+
+    const xcpdResult = outboundPatientDiscoveryRespSuccessfulSchema.safeParse(response);
+    if (!xcpdResult.success) {
+      throw new Error("Failed to parse response");
+    }
+    if (xcpdResult.success) {
```

Comment:
I uniformly try to never cast anything, which is why I was using the schema to validate to get past the typescript compiler. Also casting using OutboundPatientDiscoveryRespSuccessfulSchema is also a dependency, and from the exact same library, so thats not preferable in my view. 

Instead of throwing an error, I will just do this instead:

```
    expect(xcpdResult.success).toBe(true);
```

Reply:
Wait, but `processXCPDResponse` already returns a typed response, why would we need to parse it? This seems like an overkill, we should trust the types for code we control.